### ISSUE TITLE
Flag two opposite conditions next to each other

### ISSUE THREAD

**Issue by filiph** [dart-lang/linter#175]
``` dart
if (str[i] == '"') {
  while (i < str.length && str[i] != '"') {
    i += 1;
  }
}
```

The intent of this code is: if you encounter a double quote, skip all characters until you encounter another one.

It should be clear to the static analyzer that the while loop can never execute. (There should be another `i += 1` before the while loop.)

**Comment by alexeieleusis** [comment]
This should be linted now that https://github.com/dart-lang/linter/pull/229 is merged. I think it is not currently but fixing that should be easy. I might add a test to confirm @pq can you assign this to me?

**Comment by pq** [comment]
> I might add a test to confirm @pq can you assign this to me?

Once you accept the collaborator invite, you should be able to self-assign...

**Comment by filiph** [comment]
Is this linted? If so, we can close this bug.

Ideally, add the name of the lint here so people can find it through this bug.

**Comment by alexeieleusis** [comment]
This should be linted by `invariant_booleans`.

**Comment by pq** [comment]
@alexeieleusis : is there a test case that confirms it?

**Comment by alexeieleusis** [comment]
Sorry, I forgot to point and close, this verifies it https://github.com/dart-lang/linter/blob/master/test/rules/invariant_booleans.dart#L237

**Comment by pq** [comment]
Cool.  Thanks @alexeieleusis ! Seems like this is safe to close?

**Comment by alexeieleusis** [comment]
I just saw the extra mutation condition, previous one with this https://github.com/dart-lang/linter/blob/master/test/rules/invariant_booleans.dart#L237 should provide the desired behavior.
